src/main/java/com/redhat/pantheon/model/api/Accessor.java

```diff
@@ -0,0 +1,12 @@
+package com.redhat.pantheon.model.api;
+
+import javax.annotation.Nonnull;
+import java.util.function.Supplier;
+
+interface Accessor<T> extends Supplier<T>, ResourceMember {
```

Comment:
Here is my rationale: Is an `Accesor` always a `ResourceMember`? In the context of the api the answer should be yes because I want accessors to aways be used as such. If not, the API leaves the door open to `Accessors` which are not `ResourceMember`s. And while that is not a problem, it's not how I intend the API to be used.
Having said that, since `ResourceMember` is a marker interface, I have no problem about its intended use being directly required on the marked classes. So, all I'm saying is I'm fine either way, so if my arguments didn't convince you I can make the change.

Reply:
Actually... take a look at `ModuleDataRetriever.java:96`. We are injecting a few "unpersisted" values into the `Resources` before ultimately returning them to the client. If we were to rework that code to use your new `Module` API, what would that look like? Might that be a case where those values are represented as Fields that are both `Accessors` and `Mutators` but not `ResourceMembers`?

Reply:
Well, in that code you are not injecting values into the `Resource` object. You are in fact mapping the `Resource` to a `Map`, and then modifying said map to add some custom entries. At that point, the `SlingResource` would no longer be used.

If I were to use the new API it would be something like this:

```java
        // Get the results as Modules
        return results.map(r -> r.adaptTo(Module.class))
                // turn each module to a map
                .map(slingResource -> slingResource.toMap())
                // add the necessary extra fields to the map
                .peek(m -> m.put("extraKey", "extraValue"))
                // transform to a list
                .collect(Collectors.toList());
```

Reply:
Fair enough. I think your point is valid in that we'd be complicating the system to try and migrate away from Map for no real win.

Ok, I still think this is a bit eyebrow-raising but if you feel strongly that this is what you intend, then I defer to your judgment. There is no functional difference and if one ever surfaces in the future then we can revisit the topic then. Thanks for the healthy debate! :)